### PR TITLE
wezterm: unstable-2020-11-22 -> 20210407-nightly

### PR DESCRIPTION
**Pull request by pinpox**
###### Motivation for this change
Fixes wezterm build + annoying font bug.

@SuperSandro2000 
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by pinpox** [comment]
I still get an error on this:
```
$ nix --experimental-features nix-command build --no-link --keep-going --option build-use-sandbox relaxed -f /home/pinpox/.cache/nixpkgs-review/pr-118712/build.nix
error: builder for '/nix/store/2hw8yj4vljn7s46nrsxim3i49hc0zja7-wezterm-20210407-nightly-vendor.tar.gz.drv' failed with exit code 1;
       last 10 log lines:
       > Caused by:
       >   feature `resolver` is required
       >
       >   consider adding `cargo-features = ["resolver"]` to the manifest
       > Traceback (most recent call last):
       >   File "/nix/store/4pvjb8h2cyzxbqc4angl6f4g9jnx1qxw-cargo-vendor-normalise/bin/.cargo-vendor-normalise-wrapped", line 42, in <module>
       >     main()
       >   File "/nix/store/4pvjb8h2cyzxbqc4angl6f4g9jnx1qxw-cargo-vendor-normalise/bin/.cargo-vendor-normalise-wrapped", line 17, in main
       >     assert list(data.keys()) == ["source"]
       > AssertionError
       For full logs, run 'nix log /nix/store/2hw8yj4vljn7s46nrsxim3i49hc0zja7-wezterm-20210407-nightly-vendor.tar.gz.drv'.
error: 1 dependencies of derivation '/nix/store/kk9hpwg1yi9qahwxb5ppw238g7nhv0yd-wezterm-20210407-nightly.drv' failed to build
error: 1 dependencies of derivation '/nix/store/crdw3ihn4a426w1g286dgk6lv5x18mx7-env.drv' failed to build
```

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118712` at 3c4db116 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/4996c3abb82f0656d771ffd4f670ae11">wezterm</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

**Comment by SuperSandro2000** [comment]
> I still get an error on this, any ideas?:
> 
> ```
> $ nix --experimental-features nix-command build --no-link --keep-going --option build-use-sandbox relaxed -f /home/pinpox/.cache/nixpkgs-review/pr-118712/build.nix
> error: builder for '/nix/store/2hw8yj4vljn7s46nrsxim3i49hc0zja7-wezterm-20210407-nightly-vendor.tar.gz.drv' failed with exit code 1;
>        last 10 log lines:
>        > Caused by:
>        >   feature `resolver` is required
>        >
>        >   consider adding `cargo-features = ["resolver"]` to the manifest
>        > Traceback (most recent call last):
>        >   File "/nix/store/4pvjb8h2cyzxbqc4angl6f4g9jnx1qxw-cargo-vendor-normalise/bin/.cargo-vendor-normalise-wrapped", line 42, in <module>
>        >     main()
>        >   File "/nix/store/4pvjb8h2cyzxbqc4angl6f4g9jnx1qxw-cargo-vendor-normalise/bin/.cargo-vendor-normalise-wrapped", line 17, in main
>        >     assert list(data.keys()) == ["source"]
>        > AssertionError
>        For full logs, run 'nix log /nix/store/2hw8yj4vljn7s46nrsxim3i49hc0zja7-wezterm-20210407-nightly-vendor.tar.gz.drv'.
> error: 1 dependencies of derivation '/nix/store/kk9hpwg1yi9qahwxb5ppw238g7nhv0yd-wezterm-20210407-nightly.drv' failed to build
> error: 1 dependencies of derivation '/nix/store/crdw3ihn4a426w1g286dgk6lv5x18mx7-env.drv' failed to build
> ```

Try setting ``cargoBuildFlags = [ "--features resolver" ];``.

**Comment by pinpox** [comment]
> Try setting `cargoBuildFlags = [ "--features resolver" ];`. If it can't find that feature we need rust 1.51 which is the case since [wez/wezterm@dbd7fc2](https://github.com/wez/wezterm/commit/dbd7fc28a5350b6f7a9f3a8d1d5571197df3ab80).

Added the flags, but it doesn't seem to make any difference. How would I use rust 1.51?

**Comment by pinpox** [comment]
This should probably build against https://github.com/NixOS/nixpkgs/pull/117605

**Comment by SuperSandro2000** [comment]
@pinpox we can retarget it at staging or cherry pick the commit into this PR.

File: pkgs/applications/terminal-emulators/wezterm/default.nix [hunk 1]
```diff
@@ -77,21 +81,19 @@ rustPlatform.buildRustPackage {
 
   buildInputs = runtimeDeps;
 
-  installPhase = "" + lib.optionalString stdenv.isLinux ''
+  cargoBuildFlags = [ "--features resolver" ];
+
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/wezterm/default.nix:85]: We should test if we can drop that again or it is now required.

**Comment by SuperSandro2000** [comment]
I have cleaned up the git history a bit and force pushed it. Lets wait for the next release and then we can merge this. If in this time rust 1.51 is already in master we can just retarget the PR again.

File: pkgs/applications/terminal-emulators/wezterm/default.nix [hunk 2]
```diff
@@ -51,22 +54,23 @@ let
...
     owner = "wez";
     repo = pname;
-    rev = "3bd8d8c84591f4d015ff9a47ddb478e55c231fda";
-    sha256 = "13xf3685kir4p159hsxrqkj9p2lwgfp0n13h9zadslrd44l8b8j8";
+    rev = "d2419fb99e567e3b260980694cc840a1a3b86922";
+    sha256 = "4tVjrdDlrDPKzcbTYK9vRlzfC9tfvkD+D0aN19A8RWE=";
     fetchSubmodules = true;
   };
-  cargoSha256 = "1ghjpyd3f5dqi6bblr6d2lihdschpyj5djfd1600hvb41x75lmhx";
+  cargoSha256 = "sha256-OHbWgnlul9VfbPcMdzbuRJG59+myiukkzmnWohj5v2k=";
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/wezterm/default.nix:73]: I get another cargoSha on darwin ``sha256-UaXeeuRuQk+CWF936mEAaWTjZuTSRPmxbQ/9E2oZIqg=``..

**Comment by pinpox** [comment]
@SuperSandro2000 `nixpkgs-review pr 118712` builds nicely on my end. Everything seems to be working.

**Comment by nbraud** [comment]
I can confirm this PR works: `wezterm` builds and runs correctly.

**Comment by SuperSandro2000** [comment]
> I can confirm this PR works: wezterm builds and runs correctly. You might want to reopen the PR.

The rebase action closed it and if you didn't block me I could also quote your post without copying it.

**Pull request merged**